contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -2974,41 +2998,54 @@ del_from_bbf_schema(const char *schema_name,
 {
 	Relation	bbf_schema_rel;
 	HeapTuple	tuple_bbf_schema;
-	ScanKeyData key[5];
-	TableScanDesc scan;
+	ScanKeyData scanKey[5];
+	SysScanDesc scan;
 	int16	dbid = get_cur_db_id();
 
 	bbf_schema_rel = table_open(get_bbf_schema_perms_oid(),
 									RowExclusiveLock);
-	ScanKeyInit(&key[0],
+	ScanKeyInit(&scanKey[0],
 				Anum_bbf_schema_perms_dbid,
 				BTEqualStrategyNumber, F_INT2EQ,
 				Int16GetDatum(dbid));
-	ScanKeyInit(&key[1],
+	ScanKeyEntryInitialize(&scanKey[1], 0,
 				Anum_bbf_schema_perms_schema_name,
-				BTEqualStrategyNumber, F_NAMEEQ,
+				BTEqualStrategyNumber,
+				InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_NAMEEQ,
 				CStringGetDatum(schema_name));
-	ScanKeyInit(&key[2],
+	ScanKeyEntryInitialize(&scanKey[2], 0,
 				Anum_bbf_schema_perms_object_name,
-				BTEqualStrategyNumber, F_NAMEEQ,
+				BTEqualStrategyNumber,
+				InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_NAMEEQ,
 				CStringGetDatum(object_name));
-	ScanKeyInit(&key[3],
+	ScanKeyEntryInitialize(&scanKey[3], 0,
 				Anum_bbf_schema_perms_permission,
-				BTEqualStrategyNumber, F_NAMEEQ,
+				BTEqualStrategyNumber,
+				InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_NAMEEQ,
 				CStringGetDatum(permission));
-	ScanKeyInit(&key[4],
+	ScanKeyEntryInitialize(&scanKey[4], 0,
 				Anum_bbf_schema_perms_grantee,
-				BTEqualStrategyNumber, F_NAMEEQ,
+				BTEqualStrategyNumber,
+				InvalidOid,
+				tsql_get_server_collation_oid_internal(false),
+				F_NAMEEQ,
 				CStringGetDatum(grantee));
+	scan = systable_beginscan(bbf_schema_rel,
+				get_bbf_schema_perms_idx_oid(),
+				true, NULL, 5, scanKey);
 
-	scan = table_beginscan_catalog(bbf_schema_rel, 5, key);
-
-	tuple_bbf_schema = heap_getnext(scan, ForwardScanDirection);
+	tuple_bbf_schema = systable_getnext(scan);
 
 	if (HeapTupleIsValid(tuple_bbf_schema))
 		CatalogTupleDelete(bbf_schema_rel, &tuple_bbf_schema->t_self);
 
-	table_endscan(scan);
+	systable_endscan(scan);
 	table_close(bbf_schema_rel, RowExclusiveLock);
 
 	CommandCounterIncrement();
```

Comment:
Do we even require to CommandCounterIncrement  after CatalogTupleDelete? I dont think so